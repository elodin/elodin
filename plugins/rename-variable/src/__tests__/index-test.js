import { traverse, parse } from '@elodin/core'

import renameVariable from '../index'

describe('Renaming variables', () => {
  it('should correctly rename variables', () => {
    const file = `style Button { paddingLeft: 10 paddingRight: $theme_colors_primary }`

    const ast = parse(file).ast

    expect(
      traverse(ast, [
        renameVariable({
          rename: (prop) => prop.split('_').join('.'),
        }),
      ])
    ).toMatchSnapshot()
  })
})
