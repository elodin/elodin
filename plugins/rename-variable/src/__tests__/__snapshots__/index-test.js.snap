// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renaming variables should correctly rename variables 1`] = `
Object {
  "body": Array [
    Object {
      "body": Array [
        Object {
          "comments": Array [],
          "dynamic": false,
          "property": "paddingLeft",
          "raw": false,
          "type": "Declaration",
          "value": Object {
            "negative": false,
            "type": "Integer",
            "value": 10,
          },
        },
        Object {
          "comments": Array [],
          "dynamic": true,
          "property": "paddingRight",
          "raw": false,
          "type": "Declaration",
          "value": Object {
            "environment": false,
            "type": "Variable",
            "value": "theme.colors.primary",
          },
        },
      ],
      "comments": Array [],
      "name": "Button",
      "type": "Style",
    },
  ],
  "type": "File",
}
`;
